File: GameOfLifeJava/GameOfLifeTUI.java
import java.util.Arrays;
import java.util.Date;




public class GameOfLifeTUI {
	private Grid myGrid;
	
	
	public GameOfLifeTUI(int x, int y){
		this.myGrid=new Grid(x, y);
		myGrid.GliderSetup();
	}
	
	
	public void run(){
		
		while (true){
			clearScreen();
			this.displayGrid();
			myGrid.update();
			this.pause (1);
			clearScreen();
			
		}
		
	}
	
	private void clearScreen(){
		System.out.println("\u001b[2J");
		
	}
	
	public void displayGrid(){
		System.out.println (Arrays.deepToString(myGrid.currentState));
		
	}
	
	public void pause(int seconds){
	    Date start = new Date();
	    Date end = new Date();
	    while(end.getTime() - start.getTime() < seconds * 1000){
	        end = new Date();
	    }
	}
}

File: gameOfLife/GameOfLifeTUI.py
import time
from Grid import Grid

class GameOfLifeTUI() :

    def __init__(self,xDim,yDim) :
        self.myGrid = Grid(xDim,yDim)
        self.xDim = xDim
        self.yDim = yDim
        self.myGrid.gliderSetup()

    def run(self) :
       # on purpose an endless loop
       self.clearScreen()
       while True :
          self.displayGrid()
          self.myGrid.update()
          time.sleep(1)
          self.clearScreen()

    def displayGrid(self) :
       print(self.myGrid)

    def clearScreen(self) :
       print("\033[H\033[2J")